File: emoji-library/src/main/java/com/blackhat/emoji/helper/EmojiconGridView.java
package com.blackhat.emoji.helper;

import android.app.Activity;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.GridView;

import com.blackhat.emoji.R;
import com.blackhat.emoji.emoji.Emojicon;
import com.blackhat.emoji.emoji.People;

public class EmojiconGridView {
    public View rootView;
    EmojiconsPopup mEmojiconPopup;
    EmojiconRecents mRecents;
    Emojicon[] mData;


    public EmojiconGridView(Context context, Emojicon[] emojicons, EmojiconRecents recents, EmojiconsPopup emojiconPopup, boolean useSystemDefault) {
        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Activity.LAYOUT_INFLATER_SERVICE);
        mEmojiconPopup = emojiconPopup;
        rootView = inflater.inflate(R.layout.emojicon_grid, null);
        setRecents(recents);
        GridView gridView = rootView.findViewById(R.id.Emoji_GridView);
        if (emojicons == null) {
            mData = People.DATA;
        } else {
            mData = emojicons.clone();
        }
        EmojiAdapter mAdapter = new EmojiAdapter(rootView.getContext(), mData, useSystemDefault);
        mAdapter.setEmojiClickListener(new OnEmojiconClickedListener() {

            @Override
            public void onEmojiconClicked(Emojicon emojicon) {
                if (mEmojiconPopup.onEmojiconClickedListener != null) {
                    mEmojiconPopup.onEmojiconClickedListener.onEmojiconClicked(emojicon);
                }
                if (mRecents != null) {
                    mRecents.addRecentEmoji(rootView.getContext(), emojicon);
                }
            }
        });
        gridView.setAdapter(mAdapter);
    }

    private void setRecents(EmojiconRecents recents) {
        mRecents = recents;
    }

    public interface OnEmojiconClickedListener {
        void onEmojiconClicked(Emojicon emojicon);
    }

}

File: emoji-library/src/main/java/com/blackhat/emoji/helper/EmojiconGridView.kt
package com.blackhat.emoji.helper

import android.app.Activity
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.widget.GridView

import com.blackhat.emoji.R
import com.blackhat.emoji.emoji.Emojicon
import com.blackhat.emoji.emoji.People

import java.util.Objects

open class EmojiconGridView internal constructor(context: Context, emojicons: Array<Emojicon>?, recents: EmojiconRecents, var mEmojiconPopup: EmojiconsPopup, useSystemDefault: Boolean) {
    var rootView: View
    private var mRecents: EmojiconRecents? = null
    private var mData: Array<Emojicon>? = null


    init {
        val inflater = context.getSystemService(Activity.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        rootView = Objects.requireNonNull(inflater).inflate(R.layout.emojicon_grid, null)
        setRecents(recents)
        val gridView = rootView.findViewById<GridView>(R.id.Emoji_GridView)
        if (emojicons == null) {
            mData = People.DATA
        } else {
            mData = emojicons.clone()
        }
        val mAdapter = EmojiAdapter(rootView.context, mData, useSystemDefault)
        mAdapter.setEmojiClickListener(object : OnEmojiconClickedListener {
            override fun onEmojiconClicked(emojicon: Emojicon) {
                if (mEmojiconPopup.onEmojiconClickedListener != null) {
                    mEmojiconPopup.onEmojiconClickedListener!!.onEmojiconClicked(emojicon)
                }
                if (mRecents != null) {
                    mRecents!!.addRecentEmoji(rootView.context, emojicon)
                }
            }
        })
        gridView.adapter = mAdapter
    }

    private fun setRecents(recents: EmojiconRecents) {
        mRecents = recents
    }

    interface OnEmojiconClickedListener {
        fun onEmojiconClicked(emojicon: Emojicon)
    }

}